CONTRIBUTING.md

```diff
@@ -1,55 +1,164 @@
 # Contributing
 
-The dance tool and tests run on the host; macOS, Linux, and Windows are expected to work.
-Databases under test (FerretDB and MongoDB) may be running on the host or inside Docker; Docker Compose configuration is provided for convenience but not required.
-In particular, the FerretDB development cycle (fix-compile-run-dance) is faster with it running on the host as it does not involve Docker image building or PostgreSQL restarts.
-Running FerretDB on the host is recommended for that reason.
+This document describes how to run and develop dance tools and tests.
+
+## Types of tests
+
+You can find tests in the `tests` directory.
+We support tests of different types:
+
+* `mongo-go-driver` - to test compatibility with the MongoDB Go driver;
+* `diff` - to demonstrate FerretDB-specific differences;
+* `mongo-tools` - to test compatibility with the tools (like dump and restore).
 
-## Cloning repository
+## How to roll out dance locally
+
+Clone this repository with all submodules included:
 
 ```sh
 git clone --recursive
 ```
 
-Remember to use that command to clone this repository with all submodules included.
+If you have already cloned the repo without submodules you can run this command to include required submodules:
 
 ```sh
 git submodule update --init
 ```
 
-If you've already cloned it without submodules you can use that command
-to include required submodules.
+## Local environment
+
+The dance tool and tests run on the host; macOS, Linux, and Windows are expected to work.
+Databases under test (FerretDB and MongoDB) may be running on the host or inside Docker.
+Docker Compose configuration is provided for convenience but not required.
+
+In particular, the FerretDB development cycle (fix-compile-run-dance) is faster with it running on the host
```

Comment:
The "FerretDB development cycle" does not include running dance tests.

Reply:
Then please rephrase it to highlight which dev cycle does include running dance tests. Currently it's not clear, especially for new contributors.